staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -433,6 +433,11 @@ func (e *Store) Create(ctx context.Context, obj runtime.Object, createValidation
 	if e.Decorator != nil {
 		e.Decorator(out)
 	}
+	if dryrun.IsDryRun(options.DryRun) {
```

Comment:
why is this done here and not in DryRunnableStorage#Create?

Reply:
Because then you have to carry the dry-run flag through an amazing number of validation layers. I started out that way but it spiraled out of control very quickly.